tests/OpenTK.Tests.Math/QuaternionTests.cs

```diff
@@ -7,121 +7,152 @@ namespace OpenTK.Tests.Math
+		/// Contains all tests which cover the ctor of Quaternion.
 		/// </summary>
-		/// <param name="eulerValues">euler angle values</param>
-		/// <param name="expectedResult">expected xyz component of quaternion</param>
-		/// <param name="testName">Taken from nUnit test data. Don't know how to name data driven tests for xUnit which actually works.</param>
-		[Theory]
-		[MemberData(nameof(QuaternionTestDataGenerator.SingleAxisTestCases), MemberType = typeof(QuaternionTestDataGenerator))]
-		public void CtorEulerAnglesFloat_SingleEulerAngleSet_RotateCorrectAxis(Vector3 eulerValues, Vector3 expectedResult, string testName)
+		public class Constructor_Tests
 		{
-			//Arrange + Act: Create Quaternion with "pitch/yaw/roll"
-			var cut = new Quaternion(eulerValues.X, eulerValues.Y, eulerValues.Z);
-
-			//Assert: Use helper, to check if part of the two correct axis is zero. I just want check the direction
-			Vector3 resultXYZ = cut.Xyz;
-
-			Assert.True(QuaternionTestHelper.VerifyEqualSingleDirection(resultXYZ, expectedResult));
+			/// <summary>
+			/// Checks if a single given value (either pitch, yaw or roll) get converted into correct x,y,z value of quaternion.
+			/// </summary>
+			/// <param name="eulerValues">euler angle values</param>
+			/// <param name="expectedResult">expected xyz component of quaternion</param>
+			/// <param name="testName">Taken from nUnit test data. Don't know how to name data driven tests for xUnit which actually works.</param>
+			[Theory]
+			[MemberData(nameof(QuaternionTestDataGenerator.SingleAxisTestCases), MemberType = typeof(QuaternionTestDataGenerator))]
+			public void SingleAxisAsEulerAnglesInFloatsIsConvertedToCorrectValueInQuaternion
+			(Vector3 eulerValues, Vector3 expectedResult, string testName)
+			{
+				//Arrange + Act: Create Quaternion with "pitch/yaw/roll"
+				var cut = new Quaternion(eulerValues.X, eulerValues.Y, eulerValues.Z);
+
+				//Assert: Use helper, to check if part of the two correct axis is zero. I just want check the direction
+				Vector3 resultXYZ = cut.Xyz;
+
+				Assert.True(QuaternionTestHelper.VerifyEqualSingleDirection(resultXYZ, expectedResult));
+			}
+
+			/// <summary>
+			/// Checks if a single given value (either pitch, yaw or roll) get converted into correct x,y,z value of quaternion.
+			/// </summary>
+			/// <param name="eulerValues">euler angle values</param>
+			/// <param name="expectedResult">expected xyz component of quaternion</param>
+			/// <param name="testName">Taken from nUnit test data. Don't know how to name data driven tests for xUnit which actually works.</param>
+			[Theory]
+			[MemberData(nameof(QuaternionTestDataGenerator.SingleAxisTestCases), MemberType = typeof(QuaternionTestDataGenerator))]
+			public void SingleAxisAsEulerAnglesInVector3IsConvertedToCorrectValueInQuaternion
+			(Vector3 eulerValues, Vector3 expectedResult, string testName)
```

Comment:
Unused `testName` parameter?

Reply:
Thats a leftover from my nUnit tests. I tend to give testdata at data driven tests a name, so I know which one failed exactly. 
As I pointed out in the parameter description, I found no working way to do so with xUnit. At that point, I didn't want to purge this piece information.
I will join this problem again to find a working solution, or somebody else can help me out.

Reply:
XUnit will provide the piece of data that failed for property-based tests.

Reply:
Personally, I like to see a meaningful output at the console test runner when a specific test case fails. 
I removed the unused parameter, and moved the testcase name to comments.